### DISCUSSION TITLE
Extra Long Timelapse with Start and End time

### DISCUSSION THREAD

**Discussion by gs20050101** [gopro/labs#240] [Q&A]
Hi,

I'm new to Labs and have a few questions.
I'm trying to achieve the following: Between 30 min before sunrise and 30 min after sunset, take a photo every 2 min using Photo mode. Power off between shots to save battery.

Here is a script I came up with.  
>r-1800<s1800mP!S!120N!R

My questions
- In Extra Long Timelapse labs page, it uses NQ instead of N. What's difference between N and NQ? 
- In Extra Long Timelapse labs page, it uses !1R instead of !R. Why is the "1" necessary?

TIA

**Comment by dnewman-gpsw** [comment]
Don't do that, it will not be reliable, most cameras are good for 200-400 sleep-wake cycles before something goes wrong and the script stops. For 12 hours with a 2 minute interval, that is 360 photos, so your timelapse my not last the day.  It is okay if you can only run on internal battery, as 300 photos is getting near the battery limits.  If you have external power, use the internal photo timelapse modes, like this user: https://github.com/gopro/labs/discussions/238

0) The script you suggest will not work (<s1800 is not a command, as there aren't conditionals that support r/s rise and set times.) 

With external power use:  **!r-1800NmTPp.120!S!s1800E**     Which will wake 30 minutes before sunrise, set mode Timelapse Photo, set 2min interval, start, then end 30 minutes after sunset.

With internal battery only try:  **>20:00!r-1800NmP!S!110R**   Run the script after 8pm, it will sleep until 30 minutes before sunrise, then take a photo every 2 minutes(ish) until the battery runs out (or it gets to 8pm.)

1) difference between N and NQ are explained https://gopro.github.io/labs/control/actions/
2) The '1' shows the pause cancel message for one second, allowing you a brief time to cancel the repeat.  With just !R, you might find you are removing power to cancel.

**Reply by gs20050101** [reply]
Thank you so much for detailed answer and great insights! I appreciate it.

> 0. The script you suggest will not work (<s1800 is not a command, as there aren't conditionals that support r/s rise and set times.)

I see. It's acceptable for me to enter actual start/end times, so I can work around that part.

> With external power use: **!r-1800NmPp.120!S!s1800E** Which will wake 30 minutes before sunrise, set mode Timelapse Photo, set 2min interval, start, then end 30 minutes after sunset.

Did you mean mTP? Or, perhaps mNP, because mTP only seems to support up to p.60.

> With internal battery only try: **>20:00!r-1800NmP!S!110R** Run the script after 8pm, it will sleep until 30 minutes before sunrise, then take a photo every 2 minutes(ish) until the battery runs out (or it gets to 8pm.)

Great! Thank you. But does that script still turn off gopro between shots? 

> 1. difference between N and NQ are explained https://gopro.github.io/labs/control/actions/

I had read that. But I don't understand why anyone would use "No Action until at approximately time" when you can use "No Action until exact time".

> 2. The '1' shows the pause cancel message for one second, allowing you a brief time to cancel the repeat.  With just !R, you might find you are removing power to cancel.

Got it. thanks!

**Reply by dnewman-gpsw** [reply]
Yes mTP is the command.

p.120 means 120 seconds between shots. Most GoPro's should have that mode.  NightLapse is still a good choose.

>But does that script still turn off GoPro between shots?

No, that is why it is reliable.  If you are using external power, there is not point for risking reliability.

It you tried them, !110N vs !110NQ, you would see why NQ would use less power.

**Reply by gs20050101** [reply]
> Yes mTP is the command.
> 
> p.120 means 120 seconds between shots. Most GoPro's should have that mode. NightLapse is still a good choose.

Thanks for clarification.

> > But does that script still turn off GoPro between shots?
> 
> No, that is why it is reliable. If you are using external power, there is not point for risking reliability.

What do you mean by "reliability" here? Like, gopro might not sleep exactly 110 sec? 

> It you tried them, !110N vs !110NQ, you would see why NQ would use less power.

I'll try that once the time lapse I'm running right now finishes. I'm guessing that N won't power off gopro but NQ does?

**Reply by dnewman-gpsw** [reply]
You need to experiment, many questions are answered when you see the camera's behavior.

Reliability reread: "most cameras are good for 200-400 sleep-wake cycles before something goes wrong and the script stops".  So if you put the camera to sleep between every shoot, it is way less reliable.

**Reply by gs20050101** [reply]
I'll definitely do more experiments myself. Thanks for all your help, David!
